src/mesh.cpp

```diff
@@ -204,45 +230,72 @@ void RegularMesh::get_indices_from_bin(int bin, int* ijk) const
   }
 }
 
-bool RegularMesh::intersects(Position r0, Position r1) const
+bool RegularMesh::intersects(Position& r0, Position r1, int* ijk) const
```

Comment:
It seems pretty awkward to me that this method modifies the `r0` argument. I'd prefer to see either a separate argument that returns the intersection point or multiple return values via a `std::pair`.

Reply:
Agreed that it looks a little awkward for the general use case, but it feels somewhat more natural in `RegularMesh::bins_crossed` where the old value of `r0` isn't needed after you know the intersection point.  I tried adding another `Position&` argument as the output, and it looks like it slows down the code by about 4% on the tally test we've been running.

It's worth noting that `intersects` is not used anywhere outside of mesh.cpp.  If you like, I can remove it as a method and just have it defined in mesh.cpp.  That way, anyone who wants something like `intersect` for other purposes will probably make a new method that wraps this functionality with a cleaner interface.

Reply:
Ok, I'm fine keeping it as is. No need to worry about the interface too much until someone starts adding other mesh types :)